cineast-core/src/main/java/org/vitrivr/cineast/core/iiif/imageapi/ImageInformation.java

```diff
@@ -0,0 +1,231 @@
+package org.vitrivr.cineast.core.iiif.imageapi;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+
+/**
+ * Interface defining the common functionality implemented by ImageInformation objects of every version of the Image API
+ *
+ * @author singaltanmay
+ * @version 1.0
+ * @created 13.06.21
+ */
+public interface ImageInformation {
+
+  /**
+   * @param feature String denoting a feature whose support needs to be checked
+   * @return false if server has advertised it's supported features and the doesn't include this specific feature
+   */
+  boolean isFeatureSupported(String feature);
+
+  /**
+   * @param quality String denoting a quality whose support needs to be checked
+   * @return false if server has advertised it's supported qualities and the doesn't include this specific quality
+   */
+  boolean isQualitySupported(String quality);
+
+  /**
+   * @param format String denoting a format whose support needs to be checked
+   * @return false if server has advertised it's supported formats and the doesn't include this specific format
+   */
+  boolean isFormatSupported(String format);
+
+  /**
+   * Get the actual width of the image
+   */
+  long getWidth();
+
+  /**
+   * Get the actual height of the image
+   */
+  long getHeight();
+
+  /**
+   * Get the {@link ImageApiVersion} of the ImageInformation
+   */
+  ImageApiVersion getImageApiVersion();
+
+  /**
+   * Get the max width supported by server
+   */
+  Long getMaxWidth();
+
+  /**
+   * Get the max height supported by server
+   */
+  Long getMaxHeight();
+
+  /**
+   * Get the max area supported by server
+   */
+  Long getMaxArea();
+
+
+  /**
+   * Inner class used to parse the Image Information JSON response.
+   */
+  class SizesItem {
```

Comment:
This inner class is only used to serialize and deserialize members of the ImageInformation class and thus can be declared here instead of cluttering up the project tree